### PR TITLE
Add BSD support, fixes #1

### LINKED ISSUE #1: Missing BSD support
Unfortunately the Fyne cmd `env.go` does not compile on BSD (freebsd, openbsd, netbsd) because `report.OS` does not have any BSD support.
Ideally this should be added, but at a minimum this library should probably compile on those OSs :).

### PR DESCRIPTION
**Pull request by Jacalz**
This PR adds support for building on BSD flavors and removes some duplicated code. 

Fixes #1

### CODE REVIEW

File: report/os_unix.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// +build darwin linux openbsd freebsd netbsd dragonflypackage report
```
Comment (lucor) [inline report/os_unix.go]:
```suggestion
// +build linux openbsd freebsd netbsd dragonfly

```
Small typo. Additionally linter suggests that : `+build comment must appear before package clause and be followed by a blank line`

**Review by lucor — CHANGES_REQUESTED** [review]
LGTM, thanks.
Just some small things about the `+build` comment.
What about to have only the darwin and unix files? 
Unless we need to have specific features per OS, I'd prefer to have some small code dupllication but grouped in less files, what do you think?

File: report/os_unix.go (hunk 1 above)
Comment (Jacalz) [reply report/os_unix.go]: Good catch on the issue in the end, but the darwin part was correct. Removed this files entirely and moved it into the `os_xdg` and `os_darwin` files instead.

**Comment by Jacalz** [comment]
I agree that less files is better. It is a very small piece of code also, so that proves it even more, I guess.

**Review by lucor — APPROVED** [review]

File: report/os_xdg.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// +build linux openbsd freebsd netbsd dragonfly
```
Comment (andydotxyz) [inline report/os_xdg.go:1]: This is ineffective as there is no newline after

Comment (lucor) [reply report/os_xdg.go:1]: Good catch @andydotxyz, fixed in aee5dbc
/cc @Jacalz

Comment (Jacalz) [reply report/os_xdg.go:1]: Ah, nice catch. Thanks @andydotxyz. Will get my PR updated when I get a chance.

**Pull request merged**